From: =?utf-8?q?Jonas_=C3=85dahl?= <jadahl@gmail.com>
Date: Mon, 12 Apr 2021 11:46:30 +0200
Subject: monitor-manager: Don't include generated code in header file

Meson doesn't seem to handle depending on generated headers, at least
when those headers are pulled in indirectly via another header file.

Luckily, we don't actually need to include the generated D-Bus boiler
plate in meta-monitor-manager-private.h, since the MetaMonitorManager
type no longer is based on the D-Bus service skeleton.

So, by moving the inclusion of the generated D-Bus header file into
meta-monitor-manager.c, we should hopefully get rid of the sporadic
build issues.

Bug: https://gitlab.gnome.org/GNOME/mutter/-/issues/1682
Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1819>
Origin: upstream, 40.1, commit:91117bb052ed0d69c8ea4159c1df15c814d90627
---
 src/backends/meta-monitor-manager-private.h | 6 +++---
 src/backends/meta-monitor-manager.c         | 2 ++
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/src/backends/meta-monitor-manager-private.h b/src/backends/meta-monitor-manager-private.h
index ac3bc94..ddc3a41 100644
--- a/src/backends/meta-monitor-manager-private.h
+++ b/src/backends/meta-monitor-manager-private.h
@@ -35,8 +35,6 @@
 #include "meta/display.h"
 #include "meta/meta-monitor-manager.h"
 
-#include "meta-dbus-display-config.h"
-
 #define META_MONITOR_MANAGER_MIN_SCREEN_WIDTH 640
 #define META_MONITOR_MANAGER_MIN_SCREEN_HEIGHT 480
 
@@ -98,6 +96,8 @@ struct _MetaOutputAssignment
 #define META_IS_MONITOR_MANAGER_CLASS(klass) (G_TYPE_CHECK_CLASS_TYPE ((klass),  META_TYPE_MONITOR_MANAGER))
 #define META_MONITOR_MANAGER_GET_CLASS(obj)  (G_TYPE_INSTANCE_GET_CLASS ((obj),  META_TYPE_MONITOR_MANAGER, MetaMonitorManagerClass))
 
+typedef struct _MetaDBusDisplayConfig MetaDBusDisplayConfig;
+
 G_DEFINE_AUTOPTR_CLEANUP_FUNC (MetaMonitorManager, g_object_unref)
 
 struct _MetaMonitorManager
@@ -184,7 +184,7 @@ struct _MetaMonitorManager
  */
 struct _MetaMonitorManagerClass
 {
-  MetaDBusDisplayConfigSkeletonClass parent_class;
+  GObjectClass parent_class;
 
   GBytes* (*read_edid) (MetaMonitorManager *,
                         MetaOutput         *);
diff --git a/src/backends/meta-monitor-manager.c b/src/backends/meta-monitor-manager.c
index 1beb658..548ca80 100644
--- a/src/backends/meta-monitor-manager.c
+++ b/src/backends/meta-monitor-manager.c
@@ -60,6 +60,8 @@
 #include "meta/main.h"
 #include "meta/meta-x11-errors.h"
 
+#include "meta-dbus-display-config.h"
+
 #define DEFAULT_DISPLAY_CONFIGURATION_TIMEOUT 20
 
 enum
